akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -620,6 +620,10 @@ private[remote] class Decoder(
     (logic, logic)
   }
 
+  protected def shouldCount(v: InternalActorRef): Boolean =
+    // don't count temporary refs
+    !v.path.elements.headOption.contains("temp")
```

Comment:
A path always has at least one element so `head` would be fine, but `elements` is probably too costly here.
`root` is better, but still uncertain of the performance impact of doing this for all messages.

Are these not always `PromiseActorRef`? `isInstanceOf` would be better

Reply:
@patriknw I tried matching based on instance [like here](https://github.com/akka/akka/pull/28528/files) but that doesn't appear to work since at least part of what we're counting is based on decompressed actor refs, so instance checks don't work (not sure what the type of those actor refs is, but it's not the original `PromiseActorRef`)

Reply:
@patriknw new version will try to leverage `PromiseActorRef` checking for local refs (I think this should work - need to run this later to see if it actually does). Not sure either what's the most cost-effective approach here

Reply:
ah, one of them is probably RemoteActorRef

Reply:
would an alternative be to always count them but then filter them out when publishing the tables, which is a periodic task

Reply:
that's likely the much sounder approach, I gave it a try in the latest commit